processing/src/main/java/org/apache/druid/query/expression/TimestampExtractExprMacro.java

```diff
@@ -73,38 +74,39 @@ public Expr apply(final List<Expr> args)
       throw validationFailed("unit arg must be literal");
     }
 
-    if (args.size() > 2) {
-      validationHelperCheckArgIsLiteral(args.get(2), "timezone");
+    ISOChronology chronology = null;
+    if (args.size() > 2 && args.get(2).isLiteral()) {
+      DateTimeZone timeZone = ExprUtils.toTimeZone(args.get(2));
+      chronology = ISOChronology.getInstance(timeZone);
     }
 
-    final Expr arg = args.get(0);
     final Unit unit = Unit.valueOf(StringUtils.toUpperCase((String) args.get(1).getLiteralValue()));
-    final DateTimeZone timeZone;
 
-    if (args.size() > 2) {
-      timeZone = ExprUtils.toTimeZone(args.get(2));
-    } else {
-      timeZone = DateTimeZone.UTC;
-    }
-
-    final ISOChronology chronology = ISOChronology.getInstance(timeZone);
-
-    class TimestampExtractExpr extends ExprMacroTable.BaseScalarUnivariateMacroFunctionExpr
+    class TimestampExtractExpr extends ExprMacroTable.BaseScalarMacroFunctionExpr
```

Comment:
drive by comment: because `eval` is called every row, we often will split up implementations of `Expr` when one form has a dynamic argument that needs evaluated every row, see [`TimestampFloorExprMacro`](https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/expression/TimestampFloorExprMacro.java#L52) for example. This allows cases where the argument is a constant to skip some work.

Reply:
good catch and suggestion @clintropolis followed the example and fixed in the latest commit.

Reply:
@clintropolis and @abhishekagarwal87 the unit test cases("Remote driver error: Unauthorized" and "leadership changed") failed which should be not relative to my modification.